pwa-devdocs/src/tutorials/create-taglist-component/index.md

```diff
@@ -0,0 +1,464 @@
+
+![TagList with hard coded strings](images/hard-coded-list.png)
+
+## Step 6. Connect to real product data
+
+To make this component useful, it needs to show actual product-specific data.
+Clicking each tag should display other similarly tagged items.
+
+For this tutorial, the TagList component will list the product's categories.
+
+### Edit the GraphQL query
+
+Venia's ProductFullDetail component gets the product data from the GraphQL query defined in:
+
+```text
+packages/venia-ui/lib/queries/getProductDetail.graphql
+```
+
+Edit this file and add the following:
+
+```diff
+ meta_title @include(if: $onServer)
+ # Yes, Products have `meta_keyword` and
+ # everything else has `meta_keywords`.
+ meta_keyword @include(if: $onServer)
+ meta_description @include(if: $onServer)
+ 
++categories {
++    name
++    url_path
++}
+```
+
+Appending this to the query adds a `categories` object to the returned data.
+This object contains information on the name and URL paths for each category.
+
+### Pass the categories data into the TagList component
+
+Open the file for the product detail component and update the code:
+
+```diff
+ <section className={classes.description}>
+     <h2 className={classes.descriptionTitle}>
+         Product Description
+     </h2>
+     <RichText content={productDetails.description} />
+-    <TagList tagArray={['Tag1', 'Tag2', 'Tag3']} />
++    <TagList categories={product.categories} />
+ </section>
+```
+
+This replaces the hard coded data with actual product-specific data, but
+now, the `tagArray` prop has been replaced with a `categories` prop that the component does not know how to handle.
+
+### Update TagList
+
+Open `tagList.js` and update the component to accept the new `categories` prop:
+
+```diff
+ import React from 'react';
+ 
+-import { arrayOf, string } from 'prop-types';
++import { arrayOf, shape, string } from 'prop-types';
+ 
+ // Import the previously defined Tag component
+ import Tag from './tag';
+ 
+ // This is the main tag list module
+ const TagList = props => {
+ 
+     // Destructure the props object into variables
+-    const { tagArray } = props;
++    const { categories } = props;
+ 
+-    // Convert the array of tag strings into a list of Tag components
+-    const tagList = tagArray.map(value => {
+-        return <Tag key={value} value={value} />;
++    // Convert the array of category objects into a list of Tag components
++    const tagList = categories.map(keyword => {
++        return <Tag key={keyword.name} value={keyword.name} />;
+     })
+ 
+     // Returns the list of Tag components inside a div container
+     return (<ul>{tagList}</ul>);
+ }
+ 
+ TagList.propTypes = {
+-    // tagArray is expected to be an array of strings
+-    tagArray: arrayOf(string)
++    // categories is expected to be an object with a name and url_path string properties
++    categories: arrayOf(
++       shape:({
++           name: string,
++           url_path: string
```

Comment:
I would remove this line; you don't use `url_path` in this component.

Reply:
not yet, but it eventually expects it on a later step